app/javascript/widget/helpers/campaignHelper.js

```diff
@@ -0,0 +1,41 @@
+import { triggerCampaign } from 'widget/api/campaign';
+window.campaignTimers = [];
+
+const stripTrailingSlash = ({ URL }) => {
+  return URL.replace(/\/$/, '');
+};
+
+const startCampaigns = async ({ allCampaigns, currentURL }) => {
+  // Clear all campaign timers
+  window.campaignTimers.forEach(timerId => {
+    clearTimeout(timerId);
+  });
+  // Find all campaigns that matches the current URL
+  const filteredCampaigns = allCampaigns
+    .map(item => {
+      return {
+        ...item,
+        timeOnpage: item?.trigger_rules?.time_on_page,
+        url: item?.trigger_rules?.url,
+      };
+    })
+    .filter(item => {
+      return (
+        stripTrailingSlash({ URL: item.url }) ===
+        stripTrailingSlash({ URL: currentURL })
+      );
+    });
```

Comment:
Can you please refactor this code? Ideally, move it to a method and have a test case for the same.